source/db/dffml_source_db/source.py

```diff
@@ -0,0 +1,143 @@
+import collections
+from typing import Type, AsyncIterator, List
+
+from dffml.base import config, BaseConfig
+from dffml.db.base import BaseDatabase, Condition
+from dffml.db.sqlite import SqliteDatabase
+from dffml.record import Record
+from dffml.source.source import BaseSource, BaseSourceContext
+from dffml.util.entrypoint import entrypoint
+
+
+@config
+class DbSourceConfig(BaseConfig):
+    db_implementation: str
+    db_config: any
```

Comment:
`@config` should take care of all the loading for us. So we can just put:

```suggestion
    db: BaseDatabase
```